### PR TITLE
Add warnings for non-active addresses in receive tab and address book

### PR DESCRIPTION
**Pull request by pinheadmz**
Closes https://github.com/bitcoin/bitcoin/issues/3314
Built off of https://github.com/bitcoin/bitcoin/pull/27216 (== first two commits of this PR, required core updates)

This PR adds a "warnings" column in the receive coins dialog (recent requests table) and the address book receive page. Hovering over the ⚠️ if present shows a tooltip with a list of warnings.  We also add a "warnings" field in the receive request dialog (shown after double-clicking an address in the first table) with the same message.

The warnings shown can be anything implemented in new methods `GetWarnings()` but for now the only message we show there is:

> This address should not be used. It was derived from an inactive seed, was imported, or may have been stored unencrypted.

Example screenshots:

# 1. Create new unencrypted wallet, get new receive address

# 2. Encrypt wallet

Notice how the first address is immediately flagged.

# 3. Get new receive address after encrypting wallet

New address is not flagged, it is active and safe to use.

### CODE REVIEW

File: src/qt/recentrequeststablemodel.cpp [hunk 1]
```diff
@@ -17,19 +18,28 @@
...
 #include <QLatin1Char>
 #include <QLatin1String>
 
-RecentRequestsTableModel::RecentRequestsTableModel(WalletModel *parent) :
-    QAbstractTableModel(parent), walletModel(parent)
+QString RecentRequestEntry::GetWarnings() const
+{
+    QString warnings;
+    if (!m_is_active)
+        warnings += QObject::tr("This address should not be used. It was derived from an inactive seed, was imported, or may have been stored unencrypted.\n");
```
[the remark this answers is not in the document]
Comment (pinheadmz) [reply src/qt/recentrequeststablemodel.cpp:29]: Yeah the warning is the same, but in the GUI we have these related-but-different classes `RecentRequestEntry` for one view and `AddressTableEntry` for another. I don't see a great common file in `qt/` to declare warning strings, so I'm open to suggestions or maybe I should just start a new file?

**Comment by pinheadmz** [comment]
Thanks for all the style notes @jonatack I addressed all your comments including adding a new struct for `ReceiveRequest` and renaming/qualifying the new method to `[[nodiscard]] GetAddressWarnings()`. Also addresed your nit in https://github.com/bitcoin/bitcoin/pull/27216  and rebased both PRs on master.

**Comment by pinheadmz** [comment]
I also moved the warnings column to the right, after date. I think we could just remove the title "warnings" and that column would narrow down to 1 character ...?

Screen Shot 2023-04-05 at 1 43 25 PM

**Comment by pinheadmz** [comment]
> 🐙 This pull request conflicts with the target branch and [needs rebase](https://github.com/bitcoin-core/gui/blob/master/CONTRIBUTING.md#rebasing-changes).

Thanks D-bot! rebased.

**Comment by RandyMcMillan** [comment]
Concept ACK

**Comment by luke-jr** [comment]
Concept NACK, there's no reason such an address shouldn't be used.

**Comment by hebasto** [comment]
> Built off of [bitcoin/bitcoin#27216](https://github.com/bitcoin/bitcoin/pull/27216) (== first two commits of this PR, required core updates)

While the base PR is still under reviewing, maybe convert this one to a draft?

**Review by pablomartin4btc — COMMENTED** [review]
Concept ACK

I'd prefer your [suggestion](https://github.com/bitcoin-core/gui/pull/723#issuecomment-1497880274) of removing the title of the column or even removing the column completely and put the warning icon next to the date-time in the `Date` column as [proposed](https://github.com/bitcoin/bitcoin/issues/3314#issuecomment-81567665) by @jonasschnelli.

**Comment by pinheadmz** [comment]
closing for now, #27216 needs better approach

**Pull request closed without merging**